internal/view/owner_extender.go

```diff
@@ -0,0 +1,233 @@
+package view
+
+import (
+	"errors"
+	"fmt"
+	"github.com/derailed/k9s/internal/client"
+	"github.com/derailed/k9s/internal/ui"
+	"github.com/derailed/tcell/v2"
+	"github.com/derailed/tview"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
+const selectOwnerDialogKey = "owner"
+
+// OwnerExtender adds log actions to a given viewer.
+type OwnerExtender struct {
+	ResourceViewer
+}
+
+// NewOwnerExtender returns a new extender.
+func NewOwnerExtender(v ResourceViewer) ResourceViewer {
+	o := OwnerExtender{
+		ResourceViewer: v,
+	}
+	o.AddBindKeysFn(o.bindKeys)
+
+	return &o
+}
+
+// BindKeys injects new menu actions.
+func (o *OwnerExtender) bindKeys(aa ui.KeyActions) {
+	aa.Add(ui.KeyActions{
+		ui.KeyO: ui.NewKeyAction("Show Owner", o.ownerCmd(false), true),
+	})
+}
+
+func (o *OwnerExtender) ownerCmd(prev bool) func(evt *tcell.EventKey) *tcell.EventKey {
+	return func(evt *tcell.EventKey) *tcell.EventKey {
+		path := o.GetTable().GetSelectedItem()
+		if path == "" {
+			return nil
+		}
+		if !isResourcePath(path) {
+			path = o.GetTable().Path
+		}
+		o.showOwner(o.GetTable().GVR(), path, prev, evt)
+
+		return nil
+	}
+}
+
+func (o *OwnerExtender) showOwner(gvr client.GVR, path string, _ bool, _ *tcell.EventKey) {
+	var ownerRefs []v1.OwnerReference
```

Comment:
Take a look at how the scale_extender is implemented in particular dao.AccessorFor usage in the scale method. 
This extender could simply check if the underlying dao implements the Owned interface GetOwners() map[client.GVR][]string, . If so it will return a map of gvr/instances tuples from which we can either launch the owner related view on that instance or the selection dialog if multiple owners are detected. 
This would decouple the extender from having to list out the appropriate gvrs and simplify the extender code while keeping the behavior on the dao. 

Does this make sense?

Reply:
It all makes a lot of sense, and I tried various variants of this. The one thing I am still struggling with though: How does the Pod dao know the GVR of it's owner? I tried to solve that with the `getKindInfo`. Wouldn't every dao need to solve that same problem?

Reply:
I implemented a solution that is using the `Owned` approach. `GetOwner` code is quite repetitve so I am not sure if it is actually worth duplicating it around the daos. Still, let me know what you think.

Reply:
@derailed any feedback on this?